packages/python-packages/api-stub-generator/apistub/nodes/_function_node.py

```diff
@@ -146,13 +151,26 @@ def _parse_docstring(self):
                     kw_arg = kwargs[0]
                     self.args.remove(kw_arg)
                     self.args.append(kw_arg)
+            elif [x for x in self.args if x.argname == KW_ARG_NAME]:
+                # if kw arg details are not available in docstring and function has kwarg then flag it as error
+                self.errors.append("Keyword arg info is not available. Please ensure docstring has type info for all keyword args")
+
 
     def _parse_typehint(self):
         # Parse type hint to get return type and types for positional args
         typehint_parser = TypeHintParser(self.obj)
         # Find return type from type hint if return type is not already set
+        type_hint_ret_type = typehint_parser.find_return_type()
         if not self.return_type:
-            typehint_parser.find_return_type()
+            self.return_type = type_hint_ret_type
+        elif type_hint_ret_type:
+            # Verify return type is same in docstring and typehint if typehint is available
+            short_return_type = self.return_type.split(".")[-1]
+            long_ret_type = self.return_type
+            if long_ret_type != type_hint_ret_type and short_return_type != type_hint_ret_type:
+                error_message = "Return type in type hint is not matching return type in docstring"
+                self.errors.append(error_message)
```

Comment:
It would also be an error if there is no type hint. Though maybe that's already caught elsewhere?